qt/python/mantidqt/project/recovery/projectrecovery.py

```diff
@@ -0,0 +1,471 @@
+
+        success = self.recovery_presenter.start_recovery_view(parent=self.main_window)
+
+        if not success:
+            while not success:
+                success = self.recovery_presenter.start_recovery_failure(parent=self.main_window)
+
+        pid_dir = self.get_pid_folder_to_load_a_checkpoint_from()
+        # Restart project recovery as we stay synchronous
+        self.clear_all_unused_checkpoints(pid_dir)
+        self.start_recovery_thread()
+
+    def load_checkpoint(self, directory):
+        """
+        Load in a checkpoint that was saved by project recovery
+        :param directory: The directory from which to recover
+        :return: True, when recovery fails, False when recovery succeeds
+        """
+        # Start Regen of workspaces
+        self._regen_workspaces(directory)
+
+        # Load interfaces back. This must occur after workspaces have been loaded back because otherwise some
+        # interfaces may be unable to be recreated.
+        self._load_project_interfaces(directory)
+
+    def open_checkpoint_in_script_editor(self, checkpoint):
+        self._compile_recovery_script(directory=checkpoint)
+        self._open_script_in_editor(self.recovery_order_workspace_history_file)
+
+    def _load_project_interfaces(self, directory):
+        project_loader = ProjectLoader(self.recovery_file_ext)
+        # This method will only load interfaces/plots if all workspaces that are expected have been loaded successfully
+        if not project_loader.load_project(directory=directory, load_workspaces=False):
+            logger.error("Project Recovery: Not all workspaces were recovered successfully, any interfaces requiring "
+                         "lost workspaces are not opened")
+
+    def _regen_workspaces(self, directory):
+        self._compile_recovery_script(directory)
+
+        # Open it in the editor and run it
+        self._open_script_in_editor(self.recovery_order_workspace_history_file)
+        self._run_script_in_open_editor()
+
+    def _compile_recovery_script(self, directory):
+        alg_name = "OrderWorkspaceHistory"
+        alg = AlgorithmManager.createUnmanaged(alg_name, 1)
+        alg.initialize()
+        alg.setChild(True)
+        alg.setLogging(False)
+        alg.setRethrows(True)
+        alg.setProperty("RecoveryCheckpointFolder", directory)
+        alg.setProperty("OutputFilePath", self.recovery_order_workspace_history_file)
+        alg.execute()
+
+    def _open_script_in_editor(self, script):
+        # Get number of lines
+        with open(script) as f:
+            num_lines = len(f.readlines())
+
+        self._open_script_in_editor_call(Qt.AutoConnection, script)
+
+        # Force program to process events
+        QApplication.processEvents()
+
+        self.recovery_presenter.connect_progress_bar_to_recovery_view()
+        self.recovery_presenter.set_up_progress_bar(num_lines)
+
+    def _open_script_in_editor_call(self, connection_type, script):
```

Comment:
Does this need the connection type as a parameter?

Reply:
This is part of legacy code and is no longer needed so I have removed it.